x-pack/plugins/security_solution/public/common/components/grouping/container/index.tsx

```diff
@@ -0,0 +1,193 @@
+            }>
+          | undefined;
+      }
+    >;
+  groupPanelRenderer?: (fieldBucket: RawBucket) => JSX.Element | undefined;
+  groupsSelector?: JSX.Element;
+  inspectButton?: JSX.Element;
+  pagination: {
+    pageIndex: number;
+    pageSize: number;
+    onChangeItemsPerPage: (itemsPerPageNumber: number) => void;
+    onChangePage: (pageNumber: number) => void;
+  };
+  renderChildComponent: (groupFilter: Filter[]) => React.ReactNode;
+  selectedGroup: string;
+  takeActionItems: (groupFilters: Filter[]) => JSX.Element[];
+  unit?: (n: number) => string;
+}
+
+const GroupingContainerComponent = ({
+  badgeMetricStats,
+  customMetricStats,
+  data,
+  groupPanelRenderer,
+  groupsSelector,
+  inspectButton,
+  pagination,
+  renderChildComponent,
+  selectedGroup,
+  takeActionItems,
+  unit = defaultUnit,
+}: GroupingContainerProps) => {
+  const [trigger, setTrigger] = useState<
+    Record<string, { state: 'open' | 'closed' | undefined; selectedBucket: RawBucket }>
+  >({});
+
+  const groupsNumber = data?.groupsNumber?.value ?? 0;
+  const unitCountText = useMemo(() => {
+    const count =
+      data?.alertsCount?.buckets && data?.alertsCount?.buckets.length > 0
+        ? data?.alertsCount?.buckets[0].doc_count ?? 0
+        : 0;
+    return `${count.toLocaleString()} ${unit && unit(count)}`;
+  }, [data?.alertsCount?.buckets, unit]);
+
+  const unitGroupsCountText = useMemo(
+    () => `${groupsNumber.toLocaleString()} ${GROUPS_UNIT(groupsNumber)}`,
+    [groupsNumber]
+  );
+
+  const groupPanels = useMemo(
+    () =>
+      data.stackByMultipleFields0?.buckets?.map((groupBucket) => {
+        const group = isArray(groupBucket.key) ? groupBucket.key[0] : groupBucket.key;
+        const groupKey = `group0-${
+          isArray(groupBucket.key) ? groupBucket.key[0] : groupBucket.key
+        }`;
+
+        return (
+          <span key={groupKey}>
+            <GroupPanel
+              extraAction={
+                <GroupStats
+                  bucket={groupBucket}
+                  takeActionItems={takeActionItems(createGroupFilter(selectedGroup, group))}
+                  badgeMetricStats={badgeMetricStats && badgeMetricStats(groupBucket)}
+                  customMetricStats={customMetricStats && customMetricStats(groupBucket)}
+                />
+              }
+              forceState={(trigger[groupKey] && trigger[groupKey].state) ?? 'closed'}
+              groupBucket={groupBucket}
+              groupPanelRenderer={groupPanelRenderer && groupPanelRenderer(groupBucket)}
+              onToggleGroup={(isOpen) => {
+                setTrigger({
+                  // ...trigger, -> this change will keep only one group at a time expanded and one table displayed
+                  [groupKey]: {
+                    state: isOpen ? 'open' : 'closed',
+                    selectedBucket: groupBucket,
+                  },
+                });
+              }}
+              renderChildComponent={
```

Comment:
You probably could also just use this for the direction of the accordion arrow 🤷🏾